src/pages/index.js

```diff
@@ -75,30 +88,155 @@ addNewCardButton.addEventListener("click", () => {
   cardModal.open();
 });
 
+profileAvatarButton.addEventListener("click", () => {
+  avatarModal.open();
+});
+
 //FUNCTIONS:
 
 function createCard(cardData) {
-  const newCard = new Card(cardData, "#card-template", () => {
-    imagePreviewModal.open(cardData);
-  });
+  const newCard = new Card(
+    cardData, //data
+    "#card-template", //cardSelector
+    () => {
+      //handleImageClick
+      imagePreviewModal.open(cardData);
+    },
+    handleDeleteSubmit,
+    handleLikeClick
+  );
+
   return newCard.generateCard();
 }
 
 function handleProfileFormSubmit(inputValues) {
-  console.log(inputValues.name);
-  userInfo.setUserInfo({
-    nameInput: inputValues.name,
-    jobInput: inputValues.job,
-  });
-  profileModal.close();
+  //call method that updates button text
+  profileModal.showButtonLoading("Saving...");
+  api
+    .updateUserInfo(inputValues.name, inputValues.job)
+    .then((res) => {
+      //call method to change back to default here
+      userInfo.setUserInfo({
+        nameInput: inputValues.name,
+        jobInput: inputValues.job,
+      });
+      profileModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileModal.hideButtonLoading("Save");
+    });
 }
 
 function handleAddCardFormSubmit(inputValues) {
   const name = inputValues.title;
   const link = inputValues.url;
-  const cardData = { name, link };
+  cardModal.showButtonLoading("Creating...");
+  api
+    .addNewCard(name, link)
+    .then((res) => {
+      cardsSection.prependItem(createCard(res));
+      cardModal.close();
+      cardModal.modalForm.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardModal.hideButtonLoading("Create");
+    });
+}
+
+function handleEditAvatarFormSubmit(inputValues) {
+  const link = inputValues.url;
+  avatarModal.showButtonLoading("Saving...");
+  api
+    .updateAvatar(link)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+      editAvatarValidator._disableButton();
+      avatarModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarModal.hideButtonLoading("Save");
+    });
+}
 
-  cardsSection.prependItem(createCard(cardData));
-  cardModal.close();
-  cardModal.modalForm.reset();
+function handleDeleteSubmit(card) {
+  deleteConfirmModal.open();
+  deleteConfirmModal.handleDelete(() => {
+    //calls DELETE Method from API
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        deleteConfirmModal.close();
+        card.handleTrashButton();
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors